From a72b03ff1c93c8f02edb44b9107afa1cb8dfdfbd Mon Sep 17 00:00:00 2001
From: Joshua Watt <JPEWhacker@gmail.com>
Date: Mon, 27 Feb 2023 21:31:54 -0600
Subject: [PATCH] p_spec: Use signed char for array

Explicitly specify that it must be a signed char (since it has negative
numbers) for compatibility with platforms that default to an unsigned
char (e.g. ARM)

Upstream-Status: Inactive-Upstream [lastrelease: 2016]
Signed-off-by: Joshua Watt <JPEWhacker@gmail.com>
---
 src/p_spec.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/p_spec.cpp b/src/p_spec.cpp
index b49dd43..760dc31 100644
--- a/src/p_spec.cpp
+++ b/src/p_spec.cpp
@@ -1269,7 +1269,7 @@ void P_InitSectorSpecial(sector_t *sector, int special, bool nothinkers)
 		if (sector->special >= Scroll_North_Slow &&
 			sector->special <= Scroll_SouthWest_Fast)
 		{ // Hexen scroll special
-			static const char hexenScrollies[24][2] =
+			static const signed char hexenScrollies[24][2] =
 			{
 				{  0,  1 }, {  0,  2 }, {  0,  4 },
 				{ -1,  0 }, { -2,  0 }, { -4,  0 },
